In [3]:
import requests
import pandas as pd

In [151]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?year=2016&query_text=fire&$order=query_count'

In [152]:
params = {
}

In [153]:
response = requests.get(endpoint)

In [154]:
response

<Response [200]>

In [155]:
result = response.json()

1. Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?  

In [156]:
result

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'December',
  'year': '2016',
  'query_count': '38',
  'query_te

August has the most queries at 47

2. Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?  

In [162]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=10000&$where=query_count>100'

In [163]:
response = requests.get(endpoint)

In [164]:
response

<Response [200]>

In [165]:
result = response.json()
len(result)

1262

There are 1,262 months with over 100 queries

3. Make another API request that returns all the times "codes" was searched more than 100 times in a month. How many times did this occur? 

In [173]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=1500&$where=query_count>100&query_text=codes'

In [174]:
response = requests.get(endpoint)

In [175]:
response

<Response [200]>

In [176]:
result = response.json()
len(result)

56

There are 56 months where codes was searched over 100 times.

4. Make an API request that returns the entire Top 500 Monthly Searches dataset. Make a chart that shows the number of times "maps" was searched in a month across the entire time frame.

In [177]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=50000'

In [178]:
response = requests.get(endpoint)

In [179]:
response

<Response [200]>

In [180]:
result = response.json()
len(result)

43677

In [181]:
pd.json_normalize(result)

month_name  year query_count         query_text month
0        January  2014         223               ebid   NaN
1        January  2014         112      property maps   NaN
2        January  2014          97               maps   NaN
3        January  2014          90       property tax   NaN
4        January  2014          76               jobs   NaN
...          ...   ...         ...                ...   ...
43672       July  2021           8     sign ordinance     7
43673       July  2021           8               snap     7
43674       July  2021           8     small business     7
43675       July  2021           8   water department     7
43676       July  2021           8  value of property     7

[43677 rows x 5 columns]

In [182]:
socat_df = pd.json_normalize(result)

In [183]:
maps_df = socat_df[socat_df['query_text'] == 'maps']
maps_df

month_name  year query_count query_text month
2        January  2014          97       maps   NaN
503     February  2014         140       maps   NaN
1005       March  2014         144       maps   NaN
1503       April  2014         152       maps   NaN
2003         May  2014         119       maps   NaN
...          ...   ...         ...        ...   ...
39196   November  2020          86       maps    11
39688   December  2020         113       maps    12
41188      March  2021         132       maps     3
41683      April  2021         148       maps     4
42682       June  2021         151       maps     6

[82 rows x 5 columns]